frontends/election-manager/src/utils/cast_vote_record_files.ts

```diff
@@ -188,6 +201,77 @@ export class CastVoteRecordFiles {
 
     return result;
   }
+  /**
+   * Parses the CVR files from `files` to determine the number of CVR entries
+   * in the file and the number of those entries that are duplicates.
+   */
+  async parseAllFromFileSystemEntries(
+    files: KioskBrowser.FileSystemEntry[],
+    election: Election
+  ): Promise<CastVoteRecordFileData[]> {
+    const results: CastVoteRecordFileData[] = [];
+    for (const file of files) {
+      const parsedFileInfo = parseCvrFileInfoFromFilename(file.name);
+      try {
+        const importedFile = [...this.files].find((f) => f.name === file.name);
+        if (importedFile) {
+          results.push({
+            fileImported: true,
+            newCvrCount: 0,
+            importedCvrCount: importedFile.importedCvrCount,
+            scannerIds: importedFile.scannerIds,
+            exportTimestamp: importedFile.exportTimestamp,
+            isTestModeResults: this.fileMode === 'test',
+            fileContent: '',
+            name: file.name,
+          });
+        } else {
+          assert(window.kiosk);
+          const fileContent = await window.kiosk.readFile(file.path, 'utf-8');
+          const parsedFile = await this.parseFromFileContent(
+            fileContent,
+            file.name,
+            parsedFileInfo?.timestamp || new Date(file.mtime),
+            parsedFileInfo?.machineId || '',
+            parsedFileInfo?.isTestModeResults ?? false,
+            election
+          );
+          results.push(parsedFile);
+        }
+      } catch (error) {
+        continue;
+      }
+    }
+
+    return results;
+  }
+
+  /**
+   *  Builds a new `CastVoteRecordFiles` object by adding the parsed CVRs from
+   * `fileData` to those contained by this `CastVoteRecordFiles` instance.
+   */
+  async addFromFileData(
+    fileData: CastVoteRecordFileData,
+    election: Election
+  ): Promise<CastVoteRecordFiles> {
+    try {
+      assert(window.kiosk);
+      return await this.addFromFileContent(
+        fileData.fileContent,
+        fileData.name,
+        fileData.exportTimestamp,
+        election
+      );
+    } catch (error) {
+      return new CastVoteRecordFiles(
+        this.signatures,
+        this.files,
+        this.duplicateFilenames,
+        mapAdd(this.parseFailedErrors, () => fileData.name, error.message),
+        this.deduplicatedCastVoteRecords
+      );
+    }
```

Comment:
it seems like this error case is already handled within `addFromFileContent`, unless i'm reading it incorrectly